### PR TITLE
fix webhook emitted from orderUpdateShipping

### PR DESCRIPTION
**Pull request by lkostrowski**
I want to merge this change because it fixed webhook emitted from `orderUpdateShipping` to be DRAFT_ORDER_UPDATED if order was draft

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: saleor/graphql/order/mutations/order_update_shipping.py [hunk 1]
```diff
@@ -74,9 +75,16 @@ def perform_mutation(  # type: ignore[override]
                 "lines", "channel__shipping_method_listings"
             ),
         )
+
+        # Below this Order is properly typed Order, instead of Model
+        if not isinstance(order, models.Order):
+            raise Exception("Received node is not Order")
```
[the remark this answers is not in the document]
Comment (lkostrowski) [reply saleor/graphql/order/mutations/order_update_shipping.py:81]: I will do that, but this is not the same thing

```
def cast(typ, val):
    """Cast a value to a type.

    This returns the value unchanged.  To the type checker this
    signals that the return value has the designated type, but at
    runtime we intentionally don't check anything (we want this
    to be as fast as possible).
    """
    return val
```

which means it solves the typing issue, but it's not  "defensive", so it will not actually check instance.

However, I believe we trust the values here and type casting will be enough. Thanks 👍

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/order/mutations/order_update_shipping.py [hunk 2]
```diff
@@ -137,5 +145,13 @@ def perform_mutation(  # type: ignore[override]
 
         order.save(update_fields=SHIPPING_METHOD_UPDATE_FIELDS)
         # Post-process the results
-        call_order_event(manager, WebhookEventAsyncType.ORDER_UPDATED, order)
+
+        event_to_emit = (
+            WebhookEventAsyncType.DRAFT_ORDER_UPDATED
```
Comment (korycins) [inline saleor/graphql/order/mutations/order_update_shipping.py:150]: You should also emit the event for the `return` from the line 122

File: saleor/graphql/order/tests/mutations/test_order_update_shipping.py [hunk 3]
```diff
@@ -705,3 +705,60 @@ def test_order_update_shipping_triggers_webhooks(
...
+    permission_group_manage_orders.user_set.add(staff_api_client.user)
+    order = order_with_lines
+    order.base_shipping_price = zero_money(order.currency)
+    order.status = OrderStatus.DRAFT
+    order.save()
+
+    query = ORDER_UPDATE_SHIPPING_QUERY
+    order_id = graphene.Node.to_global_id("Order", order.id)
+    method_id = graphene.Node.to_global_id("ShippingMethod", shipping_method.id)
+    variables = {"order": order_id, "shippingMethod": method_id}
```
Comment (korycins) [inline saleor/graphql/order/tests/mutations/test_order_update_shipping.py:747]: Would be good to also have a test when you call with  `shippingMethod == null`

File: saleor/graphql/order/mutations/order_update_shipping.py (hunk 2 above)
Comment (lkostrowski) [reply saleor/graphql/order/mutations/order_update_shipping.py]: as discussed - it was out of the initial scope, but added + incldued in changelog as a separate change

**Review by korycins — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]
